### ISSUE TITLE
Failing to build library

### ISSUE THREAD

**Issue by i-schuetz** [KyleMayes/clang-sys#93]
I'm trying to create an Android build with cargo but it complains about a missing environment variable. The variable is set.

```
mobile_shared_rust $ echo $LLVM_CONFIG_PATH
/usr/local/opt/llvm/bin/
mobile_shared_rust $ which llvm-config
/usr/local/opt/llvm/bin/llvm-config

mobile_shared_rust $ cargo build --target=x86_64-linux-android
warning: couldn't execute `llvm-config --prefix` (error: No such file or directory (os error 2))
warning: set the LLVM_CONFIG_PATH environment variable to a valid `llvm-config` executable
```

You also seem to have the same warnings in your builds:
https://travis-ci.org/KyleMayes/clang-sys/jobs/564716309
https://ci.appveyor.com/project/KyleMayes/clang-sys-vtvy5

**Comment by KyleMayes** [comment]
`LLVM_CONFIG_PATH` should be set to the full path to a valid `llvm-config` executable so the path should include the executable itself.

This is not clear and the warning and documentation will be updated.

**Comment by i-schuetz** [comment]
It also happened when the executable was in the path! The only way I was able to fix it was by setting this variable directly in `build.rs`.

**Comment by KyleMayes** [comment]
I will look into it further when I get a chance.

**Comment by JosephCatrambone** [comment]
I'm getting a similar bug in Windows.  I have LLVM installed, but there's no llvm-config in Windows.  What's the recommended build.rs configuration in this case?

**Comment by KyleMayes** [comment]
I think the two issues are:

1. The warning should make it clear that LLVM_CONFIG_PATH is optional and
is only one way of assisting the build script in finding libclang

2. The warning shouldn't be printed unless the LLVM_CONFIG_PATH environment
variable is set or the build script fails to find libclang

In your case, you might want to use the LIBCLANG_PATH environment variable
mentioned in the README.

**Comment by KyleMayes** [comment]
This has been addressed in `v1.0.0`, now failure to execute commands (both `llvm-config` and `xcode-select`) is only reported as a Cargo warning when the build script fails to find an instance of `libclang`.
